### ISSUE TITLE
Allow for trimming the post title length

### ISSUE THREAD

**Issue by phil-sola** [WordPress/gutenberg#49748]
## What problem does this address?
As of Gutenberg 15.2 there has been a excerpt length control [added](https://make.wordpress.org/core/2023/02/24/whats-new-in-gutenberg-15-2-22-february/) which is great, however I feel we need to go one step further as well for the post title block. 

Many times for consistent design you may want to limit the length of a title, especially in a grid of posts e.g. 3 latest posts or the blog page displaying these in a grid.

You can't control how long the titles on a website will be sometimes and you end up with some titles spanning 1 line, others spanning 2, 3, 4 lines etc. 

Enabling users to choose the desired title length in the UI, just like the excerpt length, would be a fantastic addition to the editor. 

![image](https://user-images.githubusercontent.com/40669220/231424992-c420c037-e68f-4cb9-a0cc-b980b672e432.png)

This staggered look doesn't look great, and could be avoided by limiting the length of text used for the titles.

## What is your proposed solution?
I appreciate this can be achieved with JavaScript & CSS currently but it would be a great addition for the editor like with the excerpt length setting. 

Something I have done myself in the past is use something like the mb_strimwidth in php in partnership with an HTML title atttribute to specify the full title of the post. 

It would be nice to have maybe a toggle to turn on the title attribute as well which could then be used in tandem with the range control to limit the length of the title. That way users could still see the full title if they want and the design can remain uniform.

**Comment by aaronrobertshaw** [comment]
> Enabling users to choose the desired title length in the UI, **just like the excerpt length**, would be a fantastic addition to the editor.

We might need a slightly different approach to limiting the length of a Post Title. I believe the Post Excerpt length is limited by the number of words. As words can vary greatly in length and there is less tolerance in breaking to a new line, we might need to tweak the approach to limit by character.

**Comment by phil-sola** [comment]
I agree @aaronrobertshaw & I was thinking the same thing. This is exactly what the mb_strimwidth function does for me currently in classic themes. Trimming the title by character rather than words seems to make much more sense for the title. 

I'd love to see this gain more traction and make it's way in! I feel this is something that a lot of designers would benefit from.

**Comment by ntsekouras** [comment]
I'm not sure this is something we should have in core. It seems to me it's something that consumers can handle with code (css, `the_title` filter, etc..) and that is a niche feature, a bit more targeted to nested titles in post grids etc.. I'd appreciate some other opinions though. Maybe @jasmussen @Mamaduka @t-hamano ?

**Comment by jasmussen** [comment]
I'd softly lean towards plugin territory for this one. But it's not at all a strong opinion, and I'd think that the excerpt control if polished well enough should be transplantable. Just important that what it does especially if applied to a title, is paired with something that is understood by screen readers, and importantly isn't too verbose. We'd want to avoid a screen reader first reading the elided title for then to read again the full un-cropped title as an aria label, I assume that'd get noisy quick.

**Comment by Mamaduka** [comment]
I agree, this seems like a plugin territory. Post excerpts always had a trimming option, and the block editor tried to match it to the UI. Unfortunately, the excerpt also has some problems that are hard to resolve, and the editor is still using some hacks as workarounds - #53570.

I would recommend closing this as "maybe later".

**Comment by ntsekouras** [comment]
Let's close this for now then.
